src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -1,139 +1,2175 @@
+#include "ScriptedCreature.h"
+#include "TemporarySummon.h"
+#include "Unit.h"
+#include "Vehicle.h"
+
+// Worgen Renegade - 44793
+struct npc_worgen_renegade_44793 : public ScriptedAI
+{
+    npc_worgen_renegade_44793(Creature* creature) : ScriptedAI(creature) { }
+
+    enum eNPC
+    {
+        NPC_FORSAKEN_TROOPER1 = 44791,
+        NPC_FORSAKEN_TROOPER2 = 44792,
+
+        SPELL_HEARTSTRIKE = 84182,
+        SPELL_DARKENED = 73307,
+        SPELL_FLURRY_OF_CLAWS_CHANNEL = 80365,
+        SPELL_FLURRY_OF_CLAWS = 80367,
+
+        EVENT_CHECK_SHOWFIGHT = 101,
+        EVENT_SPELL_QUEUE = 102,
+        EVENT_SPELL_DAMAGE = 103,
+
+        MOVE_TO_HOMEPOSITION,
+    };
+
+    EventMap m_events;
+    bool m_isShowFight;
+
+    void Reset() override
+    {
+        m_events.Reset();
+
+        me->RemoveAura(SPELL_DARKENED);
+
+        m_isShowFight = true;
+
+        m_events.ScheduleEvent(EVENT_CHECK_SHOWFIGHT, 1s);
+    }
+
+    void SpellHit(Unit* caster, SpellInfo const* spell) override
+    {
+        if (!caster)
+            return;
+
+        if (spell->Id == SPELL_HEARTSTRIKE)
+            me->AddAura(SPELL_DARKENED, me);
+    }
+
+    void MovementInform(uint32 type, uint32 pointId) override
+    {
+        if (type == POINT_MOTION_TYPE && pointId == MOVE_TO_HOMEPOSITION)
+            m_events.ScheduleEvent(EVENT_CHECK_SHOWFIGHT, 25ms);
+    }
+
+    void JustEngagedWith(Unit* /*who*/) override
+    {
+        m_events.ScheduleEvent(EVENT_SPELL_QUEUE, 3s);
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        m_events.Update(diff);
+
+        while (uint32 eventId = m_events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_CHECK_SHOWFIGHT:
+                {
+                    if (me->IsInCombat() || me->isDead())
+                        return;
+
+                    if (me->GetDistance2d(me->GetHomePosition().GetPositionX(), me->GetHomePosition().GetPositionY()) > 10.0f)
+                    {
+                        me->GetMotionMaster()->MovePoint(MOVE_TO_HOMEPOSITION, me->GetHomePosition());
+                        return;
+                    }
+
+                    if (Creature* forsaken = me->FindNearestCreature(NPC_FORSAKEN_TROOPER1, 5.0f))
+                    {
+                        me->Attack(forsaken, true);
+                        return;
+                    }
+                    else if (Creature* forsaken = me->FindNearestCreature(NPC_FORSAKEN_TROOPER2, 5.0f))
+                    {
+                        me->Attack(forsaken, true);
+                        return;
+                    }
+                    else
+                        m_events.ScheduleEvent(EVENT_CHECK_SHOWFIGHT, 2s + 500ms);
+                    break;
+                }
+
+                case EVENT_SPELL_QUEUE:
+                {
+                    DoCastVictim(SPELL_FLURRY_OF_CLAWS_CHANNEL);
+                    m_events.ScheduleEvent(EVENT_SPELL_DAMAGE, 1s);
```

Comment:
hack. SPELL_FLURRY_OF_CLAWS_CHANNEL needs a spellscript to cast the damage spell onto its target on each periodic cycle

Reply:
Fixed.